src/mpc-hc/AppSettings.cpp

```diff
@@ -2860,6 +2860,42 @@ DVD_POSITION CAppSettings::CRecentFileListWithMoreInfo::GetCurrentDVDPosition()
     return DVD_POSITION();
 }
 
+void CAppSettings::CRecentFileListWithMoreInfo::UpdateCurrentAudioTrack(int audioIndex) {
+    size_t idx;
+    if (GetCurrentIndex(idx)) {
+        if (rfe_array[idx].AudioTrackIndex != audioIndex) {
+            rfe_array[idx].AudioTrackIndex = audioIndex;
+            WriteMediaHistoryEntry(rfe_array[idx]);

```

Comment:
Maybe add function that only updates the audioindex/subindex value? Similar to how abRepeat has such as function. That saves a lot of write actions, specially when toggling through available tracks.

Reply:
Ah good idea, I see what you mean.